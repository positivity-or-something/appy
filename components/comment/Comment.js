import React from "react";
import { Modal, View, Button, StyleSheet, TextInput } from "react-native";

const Comment = props => {

  return (
    <Modal visible={props.show}>
      <View>
        <TextInput
          style={styles.inputStyle}
          multiline={true}
          placeholder="Create a new comment"
          onChangeText={(text) => props.commentHandler(text)}
        />
        <Button title="Submit" 
        onPress={() => {
          props.toggleModal()
          props.addComment()
        }} />
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  inputStyle: {
    width: "100%"
  }
});

export default Comment;
