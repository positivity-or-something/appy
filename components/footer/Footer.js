import React, { Component } from "react";
import { View, Text } from "react-native";
import { Actions } from "react-native-router-flux";
import { Icon } from "react-native-elements";

class footer extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <View
        style={{
          height: 40,
          flexDirection: "row",
          justifyContent: "space-around",
          alignItems: "flex-end",
          backgroundColor: "#81DAF5",
          borderColor: "black",
          paddingBottom: 8
        }}
      >
        <Icon color='white' name="home" onPress={() => Actions.home()} />
        <Icon
          color='white'
          name="sentiment-satisfied"
          onPress={() => this.props.toggleModal()}
        />
        <Icon color='white' name="create" onPress={() => Actions.post()} />
      </View>
    );
  }
}

export default footer;
