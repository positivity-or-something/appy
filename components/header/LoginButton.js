import React from "react";
import { Button, StyleSheet } from "react-native";
import { Actions } from "react-native-router-flux";

const LoginButton = () => {
  return (
    <Button
      style={styles.button}
      backgroundColor="transparent"
      color="white"
      title="Login"
      onPress={() => console.log("CLICKED") || Actions.register()}
    >
      Login
    </Button>
  );
};

const styles = StyleSheet.create({
  button: {
    textAlignVertical: "bottom"
  }
});

export default LoginButton;
