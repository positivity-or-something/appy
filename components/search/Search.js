import React, { Component } from "react";
import {
  View,
  TextInput,
  Button,
  StyleSheet,
  Platform,
  Text
} from "react-native";
import axios from "axios";

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      post: [],
      words: " "
    };
    this.handleWords = this.handleWords.bind(this);
  }
  componentDidMount = () => {};

  handleWords(e) {
    this.setState({ words: e });
  }
  render() {
    let show = this.state.post.map((e, i) => {
      return <Text key={i}>{e.body}</Text>;
    });

    return (
      <View style={{ width: 250, height: 25, backgroundColor: "gray" }}>
        <TextInput
          onChangeText={text => {
            text.length > 0 ? this.props.handleAll(text) : this.props.hideSearch();
          }}
          type="text"
          fontSize={20}
          color="white"
          // value={this.state.post}
        />

        <Text>{show}</Text>
      </View>
    );
  }
}

export default Search;
