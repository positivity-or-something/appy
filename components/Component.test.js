import React from 'react';
import Comment from './comment/Comment';
import Footer from './footer/Footer'
import renderer from 'react-test-renderer';
import LoginButton from './header/LoginButton';
import Search from './search/Search';

test('renders correctly', () => {
  const tree = renderer.create(<Comment />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly1', () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly2', () => {
  const tree = renderer.create(<LoginButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly3x', () => {
  const tree = renderer.create(<Search />).toJSON();
  expect(tree).toMatchSnapshot();
});