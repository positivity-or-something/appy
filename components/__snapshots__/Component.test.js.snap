// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  visible={true}
>
  <View>
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      placeholder="Create a new comment"
      style={
        Object {
          "width": "100%",
        }
      }
    />
    <View
      accessibilityComponentType="button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`renders correctly 2`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "#81DAF5",
      "borderColor": "black",
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "space-around",
      "paddingBottom": 8,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        false,
        false,
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 24,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        false,
        false,
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 24,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        false,
        false,
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 24,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders correctly 3`] = `
<View
  accessibilityComponentType="button"
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
          Object {
            "color": "white",
          },
        ]
      }
    >
      Login
    </Text>
  </View>
</View>
`;

exports[`renders correctly 4`] = `
<View
  style={
    Object {
      "backgroundColor": "gray",
      "height": 25,
      "width": 250,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    color="white"
    fontSize={20}
    onChangeText={[Function]}
    type="text"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
</View>
`;

exports[`renders correctly1 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "#81DAF5",
      "borderColor": "black",
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "space-around",
      "paddingBottom": 8,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        false,
        false,
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 24,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        false,
        false,
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 24,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        false,
        false,
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 24,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders correctly2 1`] = `
<View
  accessibilityComponentType="button"
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
          Object {
            "color": "white",
          },
        ]
      }
    >
      Login
    </Text>
  </View>
</View>
`;

exports[`renders correctly3x 1`] = `
<View
  style={
    Object {
      "backgroundColor": "gray",
      "height": 25,
      "width": 250,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    color="white"
    fontSize={20}
    onChangeText={[Function]}
    type="text"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
</View>
`;
